homeworks/sasha.demidenko_SashaDemi/GamesProject/scripts/games.js

```diff
@@ -0,0 +1,232 @@
+        game_url: 'https://www.mmobomb.com/open/crossout',
+        genre: 'Shooter',
+        platform: 'PC (Windows)',
+        publisher: 'Targem',
+        developer: 'Gaijin',
+        release_date: '2017-05-30',
+        profile_url: 'https://www.mmobomb.com/crossout',
+    },
+];
+
+function createCardElement(game) {
+    const elementTemplate = document.querySelector('[data-type="card-template"]');
+    const elementCopy = elementTemplate.content.cloneNode(true);
+
+    const selectElement = (selector) => elementCopy.querySelector(`[data-type="${selector}"]`);
+
+    const cardsTitle = selectElement('title');
+    const cardsImg = selectElement('thumbnail');
+    const cardsDescription = selectElement('short_description');
+    const cardsGenre = selectElement('genre');
+    const cardsPlatform = selectElement('platform');
+    const cardsPublisher = selectElement('publisher');
+    const cardsDeveloper = selectElement('developer');
+    const cardsDate = selectElement('release_date');
+
+    cardsTitle.textContent = game.title;
+    cardsTitle.style.cssText = 'height: 90px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap';
+
+    cardsImg.src = game.thumbnail;
+    cardsImg.alt = `Thumbnail for ${game.title}`;
+    cardsImg.style.cssText = 'width: 90px; height: 90px; border-radius: 12px';
+
+    cardsDescription.textContent = game.short_description;
+    const descMaxLength = 100;
+    if (game.short_description.length > descMaxLength) {
+        cardsDescription.dataset.fullText = game.short_description;
+        cardsDescription.textContent = `${game.short_description.substring(0, descMaxLength)}...`;
+        cardsDescription.style.cursor = 'pointer';
+        cardsDescription.addEventListener('click', function () {
+            this.textContent = this.dataset.fullText;
+        });
+    }
+
+    cardsGenre.textContent = `Genre: ${game.genre}`;
+    cardsPlatform.textContent = `Platform: ${game.platform}`;
+    cardsPublisher.textContent = `Publisher: ${game.publisher}`;
+    cardsDeveloper.textContent = `Developer: ${game.developer}`;
+    cardsDate.textContent = `Release Date: ${game.release_date}`;
+
+    return elementCopy;
+}
+
+function cards(container, gamesList) {
+    const fragment = document.createDocumentFragment();
+
+    gamesList.forEach((game) => {
+        const card = createCardElement(game);
+        fragment.appendChild(card);
+    });
+
+    container.appendChild(fragment);
+}
+
+const newCards = document.querySelector('[data-type="cards-container"]');
+if (newCards) {
+    cards(newCards, games);
+}
+
+function filterGamesByReleaseDate(gamesList, isNewChecked, isOldChecked) {
+    if (isNewChecked && isOldChecked) {
+        return gamesList;
+    }
+
+    return gamesList.filter((game) => {
+        const releaseDate = new Date(game.release_date);
+        if ((isNewChecked && releaseDate.getFullYear() <= 2020)
```

Comment:
[DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
```suggestion
       const releaseYear = releaseDate.getFullYear()
       const isGameMatchedFiltes = 
         (isNewChecked && releaseYear >= 2020) &&
         (isOldChecked && releaseYear <= 2010)
       return isGameMatchedFiltes
```